aurci/general.py

```diff
@@ -32,10 +32,10 @@ def build_metainfo_dict():
         for repo in rosdistro:
         #Go through distro, and make entry for each package in a repository
             d = rosdistro[repo]
-            if 'source' in d:
-                src = d['source']['url']
-            elif 'release' in d:
+            if 'release' in d:
```

Comment:
The problem is that the release repos are always heavily outdated, a lot of fixes take longer and that's why we use the upstream dev repos. See this thread https://github.com/ros-arch/ros-package-tools/issues/1 or many others why ros-gbp is bad to use for Arch.

Reply:
Why should the release repositories be outdated? Outdated relative to what?

From my understanding, the intention is to mirror the state of the ubuntu binary packages, so the versions available on AUR match the versions released to the rosdistro as noted in the [distribution.yaml](https://github.com/ros/rosdistro/blob/master/melodic/distribution.yaml). As the release repositories are only managed automatically by the bloom release process, they should reflect the state being released to the respective ROS distribution.

I completely share the points raised in [this post](https://github.com/ros-arch/ros-package-tools/issues/1#issuecomment-655515559), but I also see the problem with a delayed `distribution.yaml`. However, this will probably be an issue for the time that the distribution syncs are done (which from my understanding is done once a month). I agree that this isn't optimal, but it would keep things much closer to the automated workflow. However, with the quite large list of outdated ROS packages in AUR (and not only for melodic because of a failing CI, but also noetic) this might be a sane way to go.

That being said, I also agree on [this post](https://github.com/ros-arch/ros-package-tools/issues/1#issuecomment-655173338) as a strong argument against using the release repositories.

In the end, this is only a suggestion and I would be fine not implementing this any further if the disadvantages outweigh the advantages of my proposal.

Reply:
And, as mentioned by @stertingen a newer release version could be fetched from the `tracks.yaml` file in order to circumvent the cached distribution problems. However, this might lead to failing builds, as those versions are not necessarily passing the buildfarm CI.

Reply:
> I completely share the points raised in [this post](https://github.com/ros-arch/ros-package-tools/issues/1#issuecomment-655515559), but I also see the problem with a delayed `distribution.yaml`. However, this will probably be an issue for the time that the distribution syncs are done (which from my understanding is done once a month). I agree that this isn't optimal, but it would keep things much closer to the automated workflow. However, with the quite large list of outdated ROS packages in AUR (and not only for melodic because of a failing CI, but also noetic) this might be a sane way to go.

One thing why we originally added CI was because the releases actually only work for Ubuntu/Debian and we had to find out where we would need new releases to get things working without patching everything on Arch. And the CI failed a lot because of the old releases in the melodic sync, sometimes thing even didn't went into sync. I'm working (or already finished but don't have time to test) on a big CI improvement to actually make package building more sane and clear. We started the CI/CD in first line bc of all the comments in about things not working on machines that didn't had certain deps or even weirder config stuff and it improved a lot faster bc of the CI, even though it doesn't look like that currently. You have valid reasons for ros-gbp as they existed always, but we changed from them to the upstream releases bc ros-gbp is made for Ubuntu and even there often requires lots of patches to work.